client/core/types.go

```diff
@@ -125,6 +125,19 @@ type SupportedAsset struct {
 	Info   *asset.WalletInfo `json:"info"`
 }
 
+func (sa *SupportedAsset) IsRescanner() bool {
+	if sa.Wallet == nil {
+		return false
+	}
+	walletType := sa.Wallet.WalletType
+	for _, wd := range sa.Info.AvailableWallets {
+		if wd.Type == walletType {
+			return wd.Traits&asset.WalletTraitRescanner == asset.WalletTraitRescanner
```

Comment:
For background, this method was modeled similarly to this frontend method for accessing a def from a SupportedAsset based on Type:

https://github.com/decred/dcrdex/blob/fb7bb9da37a60f7cd6f4bea7707d11c31fd77610/client/webserver/site/src/js/app.js#L683-L687

If we go with a `WalletState` field instead, this would reduce to `return sa.Wallet.Traits.IsRescanner()` or similar.

Reply:
We can definitely echo the bitfield implementation in javascript.